### PR TITLE
Use spawnSync to avoid hitting buffer limits

### PR DESCRIPTION
**Pull request by conorbranagan**
process `spawn` will read data into a stream as soon as it starts executing. This should help us avoid having to keep bumping up the buffer limit which we have hit again.

For more details see: http://www.hacksparrow.com/difference-between-spawn-and-exec-of-node-js-child_process.html

cc @LeoCavaille

### CODE REVIEW

**Comment by LeoCavaille** [comment]
Could you make the test suite pass?

**Comment by conorbranagan** [comment]
```
$ npm test
...
  93 passing (38s)
```

**Review by LeoCavaille — APPROVED** [review]

**Comment by LeoCavaille** [comment]
Out of curiosity I haven't seen another deployment crash did you get something like that?

**Comment by conorbranagan** [comment]
I updated the deployment to use an early cut of this update to unblock getting some config required for metric queries. Once this gets merged I'll fix up the image to our repo (https://github.com/DataDog/images/commit/67ea06861604d55686481943561c03c3a8d52d05)

**Pull request merged**
